embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/network/logging/EmbraceHttpPathOverride.kt

```diff
@@ -0,0 +1,65 @@
+package io.embrace.android.embracesdk.internal.network.logging
+
+import io.embrace.android.embracesdk.annotation.InternalApi
+import java.nio.charset.StandardCharsets
+import java.util.regex.Pattern
+
+/**
+ * Header used to override the URL's relative path
+ */
+const val PATH_OVERRIDE: String = "x-emb-path"
+
+/**
+ * Max length of relative path override
+ */
+private const val RELATIVE_PATH_MAX_LENGTH = 1024
+
+/**
+ * Allowed characters in relative path
+ * As per https://tools.ietf.org/html/rfc3986#section-2 and https://stackoverflow.com/a/1547940.
+ * Removed certain characters we do not want present: ?#
+ */
+private val RELATIVE_PATH_PATTERN: Pattern = Pattern.compile("[A-Za-z0-9-._~:/\\[\\]@!$&'()*+,;=]+")
+
+@InternalApi
+fun getOverriddenURLString(request: HttpPathOverrideRequest): String {
+    return getOverriddenURLString(request, request.getHeaderByName(PATH_OVERRIDE))
+}
+
+@InternalApi
+fun getOverriddenURLString(request: HttpPathOverrideRequest, pathOverride: String?): String {
```

Comment:
Is it not syntactically possible to set the default value of `pathOverride` using `request` to avoid defining 2 overloaded functions?